### PR TITLE
feat: skip db update for clientserver mode

### LINKED ISSUE #1430: Option to skip DB update in ClientServer mode
It would be great to have a possibility to skip DB update in case of using external trivy-server (ClientServer mode). Since at the moment trivy-server do not support Java DB distribution in the same way as it works for common vuln db (and [that probably won't change anytime soon](https://github.com/aquasecurity/trivy/issues/3560)), this option will allow you to use your own docker image for trivy scan jobs, which already contains updated databases, and still use external trivy-server for caching scan results.

### PR DESCRIPTION
**Pull request by ahalay**
## Related issues
- Close #1430

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/aquasecurity/trivy-operator/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy-operator/tree/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by chen-keinan** [comment]
@ahalay thank you for the contribution , I have added on comment.
also need to run `make generate-helm-docs` and `generate-all` to generate relevant manifest and docs

**Review by chen-keinan — APPROVED** [review]

File: deploy/helm/values.yaml [hunk 1]
```diff
@@ -383,6 +383,10 @@ trivy:
   serverCustomHeaders: ~
   # serverCustomHeaders: "foo=bar"
 
+  # -- clientServerSkipUpdate is the flag to enable skip databases update for Trivy client.
+  # Only applicable in ClientServer mode.
+  # clientServerSkipUpdate: true
```
[the remark this answers is not in the document]
Comment (ahalay) [reply deploy/helm/values.yaml]: changed and regenerated helm docs

**Comment by chen-keinan** [comment]
@ahalay just noticed about this change it will not solve your problem to skip java db.
 it requires a different flag `--skip-java-db-update` , you can enhance your PR to support this additional flag

**Review by chen-keinan — APPROVED** [review]

**Comment by ahalay** [comment]
@chen-keinan Does it make sense to add `--skip-java-db-update` for `SkipDBUpdate` to already existing `--skip-db-update`?
https://github.com/aquasecurity/trivy-operator/blob/d2825cd290de91bf702c2972d12f6bfa34ef1a59/pkg/plugins/trivy/flags.go#L60

**Comment by chen-keinan** [comment]
`--skip-java-db-update` refer to java component index data not vulnerability.
it occur only when trivy cli detect jar files inside to container.
 it functionality should be in a separate flag as it does a different thing from `db-update`

**Pull request merged**